syntax = "proto3";

package notepad;

message User {
  string name = 1;
  string email = 2;
  string password = 3;
  string location = 4;
}

message Users { repeated User users = 1; }

message Content { string line = 1; }

message Notepad {
  string id = 1;
  string name = 2;
  string user_name = 3;
  repeated Content contents = 4;
}

enum ChangeName {
  ADD = 0;
  REMOVE = 1;
  UPDATE = 2;
}

message Change {
  ChangeName name = 1;
  int32 start_line = 2;
  int32 end_line = 3;
  repeated Content contents = 4;
}

message Version {
  string id = 1;
  int32 number = 2;
  string created_time = 3;
  string notepad_id = 4;
  repeated Change changes = 5;
}

message Versions { repeated Version versions = 1; }